src/normal.cc

```diff
@@ -189,9 +189,24 @@ String build_autoinfo_for_mapping(const Context& context, KeymapMode mode,
             descs.emplace_back(std::move(keys), built_in.docstring);
     }
 
+    // To combine mapped keys with the same command and docstrings (synomyms),
+    // we need a map (command, docstring) -> mapped keys
+    HashMap<std::pair<String, String>, Vector<String>> synonyms;
     for (auto& key : keymaps.get_mapped_keys(mode))
-        descs.emplace_back(key_to_str(key),
-                           keymaps.get_mapping(key, mode).docstring);
+    {
+        KeymapManager::KeymapInfo key_info = keymaps.get_mapping(key, mode);
+        std::pair<String, String> cmd_and_docstr = {join(key_info.keys | transform(key_to_str)),
+                                                    key_info.docstring};
+        if (synonyms.contains(cmd_and_docstr)) {
+          synonyms[cmd_and_docstr].push_back(key_to_str(key));
```

Comment:
is there a reason you did not use `emplace_back` here ?

Reply:
I thought `emplace_back` was useful for forwarding params to a constructor, why should I prefer it in this case?

Reply:
Both are fine by me, I would tend to use push_back in the case where emplace_back would call the copy constructor anyway, such as here.